### PR TITLE
Remove dep on `fraction`

### PR DESCRIPTION
**Pull request by Manishearth**
See discussion in https://github.com/unicode-org/icu4x/pull/3922#discussion_r1355488989

Doesn't seem too complicated.

cc @younies @robertbastian

### CODE REVIEW

File: provider/datagen/src/transform/cldr/units/helpers.rs [hunk 1]
```diff
@@ -281,9 +281,7 @@ pub fn process_constants<'a>(
 /// - " 1.5 E -2 " is converted to 15/1000
 /// - " 1.5 E - 2" is an invalid scientific notation number
 /// - "1.5E-2.5" is an invalid scientific notation number
-pub fn convert_scientific_notation_to_fraction(
-    number: &str,
-) -> Result<GenericFraction<BigUint>, DataError> {
+pub fn convert_scientific_notation_to_fraction(number: &str) -> Result<Ratio<BigInt>, DataError> {
```
Comment (younies) [inline provider/datagen/src/transform/cldr/units/helpers.rs:284]: why not using [`BigRational`](https://docs.rs/num/latest/num/type.BigRational.html)

**Review by younies — COMMENTED** [review]
@Manishearth , have you considered using [BigRational](https://docs.rs/num/latest/num/type.BigRational.html)

File: provider/datagen/src/transform/cldr/units/helpers.rs (hunk 1 above)
Comment (Manishearth) [reply provider/datagen/src/transform/cldr/units/helpers.rs:284]: There's no need to add a dep on the `num` metacrate for this, and BigRational is just a typedef

Comment (Manishearth) [reply provider/datagen/src/transform/cldr/units/helpers.rs:284]: We're trying to keep dependencies low; I'm fine pulling in more `num` crates but we don't have to pull in _all_ of them.

Comment (younies) [reply provider/datagen/src/transform/cldr/units/helpers.rs:284]: Okay, got it

**Review by younies — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**